File: Hackerrank/Problem Solving/Cat and Mouse.cpp
#include<iostream>
#include<stdlib.h>
using namespace std;
int main()
{
    int q,x,y,z,i,a[100][3],d1,d2;
    cin>>q;

    for(i=0;i<q;i++)
    {
        cin>>x>>y>>z;
        a[i][0]=x;
        a[i][1]=y;
        a[i][2]=z;
    }

    for(i=0;i<q;i++)
    {
        d1=abs(a[i][0]-a[i][2]);
        d2=abs(a[i][1]-a[i][2]);
        if(d1<d2)
        cout<<"Cat A"<<endl;
        else if(d2<d1)
        cout<<"Cat B"<<endl;
        else if(d1==d2)
        cout<<"Mouse C"<<endl;
    }

}

File: Hackerrank/Problem Solving/Cat and Mouse.py
def catnMouse(tc):  
    for i in range(tc):
        c1,c2,m=map(int,input().split())
        if(abs(c1-m)<abs(c2-m)):
            print("Cat A")
        elif(abs(c1-m)>abs(c2-m)):
            print("Cat B")
        else:
            print("Mouse C")

tc=int(input("Enter the no. of testcases:"))
catnMouse(tc)
